stream-chat-android-ui-uitests/src/androidTest/java/io/getstream/chat/android/uitests/ui/action/WaitViewAction.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.uitests.ui.action
+
+import android.view.View
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.ViewInteraction
+import androidx.test.espresso.matcher.ViewMatchers.isRoot
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.util.TreeIterables
+import org.hamcrest.Matcher
```

Comment:
For some reason, registering idling resource for coroutine dispatcher is not enough and the test is still flaky.

Reply:
Might depend on the dispatcher we're using - it should be Unconfined/Main.immediate or test dispatcher I guess.

Reply:
Changed to `Main.immediate` dispatcher.